classes/PodsUI.php

```diff
@@ -5088,50 +5112,38 @@ public function restricted( $action = 'edit', $row = null ) {
 				}//end if
 			}//end foreach
 
-			if ( ! empty( $author_restrict ) ) {
-				if ( is_object( $this->pod ) && 'manage' === $action ) {
-					if ( ! in_array( 'edit', $this->actions_disabled ) && ! current_user_can( 'pods_edit_' . $this->pod->pod ) && ! in_array( 'delete', $this->actions_disabled ) && ! current_user_can( 'pods_delete_' . $this->pod->pod ) ) {
-						$okay = false;
-					}
-				}
-				if ( is_object( $this->pod ) && ! current_user_can( 'pods_' . $action . '_' . $this->pod->pod ) ) {
-					$okay = false;
-				}
-				/*
-				@todo determine proper logic for non-pods capabilities
-				elseif ( !current_user_can( 'pods_' . $action . '_' . $_tbd ) )
-				$okay = false;
-				*/
-
-				if ( ! $okay && ! empty( $row ) ) {
-					foreach ( $this->restrict['author_restrict'] as $key => $val ) {
-						$author_restricted = $this->get_field( $key );
-
-						if ( ! empty( $author_restricted ) ) {
-							if ( ! is_array( $author_restricted ) ) {
-								$author_restricted = (array) $author_restricted;
-							}
+			// Legacy author restrict check.
+			if ( $author_restrict && ! $okay && ! empty( $row ) ) {
+				foreach ( $this->restrict['author_restrict'] as $key => $val ) {
+					$author_restricted = $this->get_field( $key );
 
-							if ( is_array( $val ) ) {
-								foreach ( $val as $v ) {
-									if ( in_array( $v, $author_restricted ) ) {
-										$okay = true;
-									}
+					if ( ! empty( $author_restricted ) ) {
+						if ( ! is_array( $author_restricted ) ) {
+							$author_restricted = (array) $author_restricted;
+						}
+
+						if ( is_array( $val ) ) {
+							foreach ( $val as $v ) {
+								if ( in_array( $v, $author_restricted ) ) {
```

Comment:
:warning: Not using strict comparison for in_array; supply true for third argument.

<sub>hash: 59f60b68f4e03c4c915286329994a08a</sub>